frappe/email/doctype/email_queue/email_queue.py

```diff
@@ -30,6 +31,17 @@ def get_duplicate(self, recipients):
 		duplicate.set_recipients(recipients)
 		return duplicate
 
+	def update_db(self, commit=False, **kwargs):
+		frappe.db.set_value(self.DOCTYPE, self.name, kwargs)
+		if commit:
+			frappe.db.commit()
+
+	def update_status(self, status, commit=False, **kwargs):
+		self.update_db(status = status, commit = commit, **kwargs)
+		if self.communication:
+			communication_doc = frappe.get_doc('Communication', self.communication)
+			communication_doc.set_delivery_status(commit=commit)
+
```

Comment:
Hi,
Thanks for manually backporting
Why do we need to add these two methods?

email_queue.py in version-13-hotfix and considerably different from the same file in develop. So moving all things here might not be necessary

Reply:
> Why do we need to add these two methods?

The mail app handles the queue status update via mailgun webhooks. Since nightly already handles it, I thought it would've been easier if v13 and nightly had similar pattern without refactoring the mail app status handling code.

> email_queue.py in version-13-hotfix is considerably different from the same file in develop. So moving all things here might not be necessary

Yeah maybe I should drop this and handle it manually in mail app